_user/rotrectangle/main.c

```diff
@@ -264,14 +256,23 @@ int main(void)
 	rotrectangle.maxyc = graph.height - rotrectangle.minyc;
 	rotrectangle.maxxc = graph.width - rotrectangle.minxc;
 
-	ret = beginthread(rotrectangle_getkey, 4, stack, sizeof(stack), &rotrectangle);
+	rotrectangle_switchmode(0);
+
+	if ((stack = (char *)malloc(stacksz)) == NULL)
+		return 1;
 
-	ret = rotrectangle_rotating(&rotrectangle, &graph);
+	mutexCreate(&rotrectangle.lock);
+
+	beginthread(rotrectangle_getkey, 4, stack, stacksz, (void *)&rotrectangle);
+
+	rotrectangle_rotating(&rotrectangle, &graph);
+
+	rotrectangle_switchmode(1);
 
 	graph_close(&graph);
 	graph_done();
 
 	free(stack);
```

Comment:
I don't know if we have `join` in our native api, but it could be that we would free the stack of still-running thread